lib/rules/no-useless-constructor.js

```diff
@@ -25,13 +25,19 @@ module.exports = function(context) {
             return false;
         }
 
-
         return body[0].expression.arguments.every(function(arg, index) {
-            return (arg.type === "Identifier" && arg.name === ctorParams[index].name) ||
-            (
+            return (
+                arg.type === "Identifier" &&
+                ctorParams.length &&
+                arg.name === ctorParams[index].name
+            ) || (
                 arg.type === "SpreadElement" &&
+                ctorParams.length &&
                 ctorParams[index].type === "RestElement" &&
                 arg.argument.name === ctorParams[index].argument.name
+            ) || (
+                arg.type === "SpreadElement" &&
+                arg.argument.name === "arguments"
```

Comment:
@michaelficarra thanks for the examples! So you are saying we should only warn on this iff `constructor` doesn't receive any named/defined arguments and super is called with `...arguments`, correct?